ansible_collections/arista/avd/roles/eos_designs/templates/core_interfaces/logic.j2

```diff
@@ -0,0 +1,119 @@
+{# Parse core_interfaces data model #}
+{% set core_interfaces_data.p2p_interfaces = {} %}
+{% set core_interfaces_data.p2p_port_channels = {} %}
+{% set core_interfaces_data.bgp_neighbors = {} %}
+{% set core_interfaces_data.ospf_interfaces = [] %}
+{% set core_interfaces_data.isis_interfaces = [] %}
+{% for p2p_link in core_interfaces.p2p_links | arista.avd.natural_sort %}
+{%     if p2p_link.profile is arista.avd.defined and core_interfaces.p2p_links_profiles is arista.avd.defined %}
```

Comment:
Just a suggestion: I believe it's not required to do a defined test on **core_interfaces.p2p_links_profiles** and then pass default filter over **core_interfaces.p2p_links_profiles[p2p_link.profile]**

Simply we can do something like below:
```
{%     if p2p_link.profile is arista.avd.defined and core_interfaces.p2p_links_profiles[p2p_link.profile] is arista.avd.defined %}
{%         set p2p_profile = core_interfaces.p2p_links_profiles[p2p_link.profile]  %}
```

Reply:
if nothing is found then it will go to else. and set {}

Reply:
Have you tested that method if `core_interfaces.p2p_links_profiles` is not defined? I believe it will trigger an `undefined variable` error because of the `[ ]`.
Emil just copied the logic we have in `l3_edge`.

Reply:
I tested this on Ansible templates site. I will check it now.

Reply:
I didn't define core_interfaces (in CORE_UNIT_TESTS.yml) and didn't get any error.

Reply:
We will let this pass. In interest of time.